projects/RabbitMQ.Client/client/impl/AutorecoveringConnection.Recovery.cs

```diff
@@ -158,6 +161,11 @@ private bool TryPerformAutomaticRecovery()
             }
             catch (Exception e)
             {
+                if(connectionRecoverySucceeded)
+                {
+                    this.Close(506, "Recovery failed.", true);
```

Comment:
Note that RabbitMQ won't do anything special about a `5xx` response (besides logging the error message).

Reply:
I looked up the code's in the AMQP spec and think this was the most matching one. Not a specific reason for it.

Reply:
This is not particularly specific. Should it say "channel recovery failed"?

Reply:
No it is a general issue kicking in when exception is catched. At this point we don't know i think or we should look at the exception.